### PR TITLE
Fix `safeCopy` to work on Linux

### PR DESCRIPTION
**Pull request by opotowsky**
## What is the change?

Replace `shutil.copyfile` and `shutil.copymode` with a `copy` (windows) or `cp` (linux) command. This accomplishes the same thing as before (file copy with metadata). 

We are NOT taking away the fact that `safeCopy` overrides `shutil.copy` in this PR, because there are significant downstream impacts and we need to check in with other ARMI users first. That is being tracked in the ticket #1690.

## Why is the change being made?

Depending on how permissions are set up, this doesn't work on all linux systems.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by opotowsky** [comment]
More internal testing/processes are needed before this can be merged, but it's ready for a review, at least

File: armi/utils/__init__.py
Comment (john-science) [inline armi/utils/__init__.py:815]: Personal preference. I don't like historical notes in the code. That's what commits and PRs are for.

When I'm looking at the code, I am trying to understand what it is doing, and I would rather have a comment that describes THAT.

Comment (opotowsky) [reply armi/utils/__init__.py:815]: Will fix!

File: armi/utils/__init__.py
Comment (opotowsky) [inline armi/utils/__init__.py:821]: note that we need an exit strategy here. Decide on a max time then throw an exception?

File: armi/utils/__init__.py
Comment (opotowsky) [reply armi/utils/__init__.py]: done

File: armi/utils/__init__.py
Comment (opotowsky) [reply armi/utils/__init__.py:829]: done

File: armi/utils/__init__.py
Comment (opotowsky) [reply armi/utils/__init__.py]: done

File: armi/utils/__init__.py
Comment (opotowsky) [inline armi/utils/__init__.py:811]: Hey @john-science I wanted to check with you on this choice. It's needed since the file paths were used to being handled via python and so going straight to the command line broke some file copies.

Comment (john-science) [reply armi/utils/__init__.py:811]: Changing everything to absolute paths is generally a good idea and I like it.

File: armi/utils/__init__.py
Comment (john-science) [inline armi/utils/__init__.py:838]:
```suggestion
                + f"a maximum wait time of {maxWaitTime/60} minutes."
```

Comment (john-science) [reply armi/utils/__init__.py:838]: Just minor phrasing.  I like log statements to be more "language-ish" than "code-ish".  But I won't die on this hill.

File: armi/utils/__init__.py
Comment (john-science) [inline armi/utils/__init__.py:827]: The "else" case here will raise an error because the `cmd` variable doesn't exist yet.

Comment (opotowsky) [reply armi/utils/__init__.py:824]: Oh, oops. 

This begs the question:
Do we want a ton of "if windows elif linux else kill the run" all throughout the code base? Why not just kill the run from the getgo?

Comment (john-science) [reply armi/utils/__init__.py:824]: Well, we need something here, otherwise this is just invalid Python.  Hmmm... I will make a recommend.

File: armi/utils/__init__.py
Comment (john-science) [inline armi/utils/__init__.py:820]:
```suggestion
        raise OSError(
```

File: armi/utils/__init__.py
Comment (jakehader) [inline armi/utils/__init__.py:813]: Can we wrap this in a try/except and then proceed with the Windows / Linux check? Additionally, the detection of Windows / Linux and any language specific utility code requirements would be helpful at the start on ARMI. Is it possible to search the code base for these (maybe in a couple places where we use if "Win..." or if "Linux" and then pull those checks to ARMI __init__ or bootstrap? Just thinking about a potential speed up for failure if something is missing

Comment (opotowsky) [reply armi/utils/__init__.py:813]: I don't like keeping code in that is that fragile. I think the change I made is a little more robust

As far as the OS-check, I agree we shouldn't be adding a gazillion checks throughout the code. We should catch the OS failure during init or bootstrap like you suggest, then we can safely do "if win this else that" throughout the code.

But @john-science gets the final call there.

Comment (opotowsky) [reply armi/utils/__init__.py:813]: Actually @jakehader: I can have the windows `if` do the old/shutil approach, and the linux `if` do the `cp`/command line stuff. I'm fine with that. We know it works on windows and has for years.

Comment (opotowsky) [reply armi/utils/__init__.py:813]: @jakehader for the OS check, we are going to handle that in a different ticket. The code for now should stay until that ticket is addressed. It's out of scope for this PR.

And for windows, @john-science prefers the `copy` approach over the `shutil` approach. I don't care which we go with, but it seems cleaner to me to have parallel approaches implemented, so I lean towards the way I have it coded now. 

So as of right now John is just awaiting me to complete some internal testing to prove this PR is ready.

Comment (john-science) [reply armi/utils/__init__.py:813]: @opotowsky @jakehader That's a nice idea, I like it.  But I decided to make that it's own [ticket](https://github.com/terrapower/armi/issues/1693).

This is for Arrielle's expediency right now.  But more so, I like to keep "one feature = one PR" when I can.

Comment (opotowsky) [reply armi/utils/__init__.py:813]: HAHA @john-science jinx
https://github.com/terrapower/armi/issues/1694

Comment (opotowsky) [reply armi/utils/__init__.py:813]: feel free to close my duplicate ticket

**Review by john-science — APPROVED** [review]

**Pull request merged**
